cmds/dhclient/dhclient.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"time"
+
+	"github.com/d2g/dhcp4"
+	"github.com/d2g/dhcp4client"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	defaultIface = "eth0"
+	// slop is the slop in our lease time.
+	slop = 10
+)
+
+var (
+	iList        = []string{defaultIface}
+	renewals     = flag.Int("renewals", 2, "Number of DHCP renewals before exiting")
+	verbose      = flag.Bool("verbose", false, "Verbose output")
+	debug        = func(string, ...interface{}) {}
+	leasetimeout = flag.Int("timeout", 600, "Lease timeout in seconds")
+)
+
+/*
+ * Example Client
+ */
+func dhclient(ifname string, timeout time.Duration, done chan error) {
+	var err error
+
+	// if timeout is < 10 seconds, let's get real.
+	if timeout < slop {
```

Comment:
time.Duration is an int64 that's measured in nanoseconds. So this line is "if timeout < 10ns" and then sets the timeout to 20ns. I think you want

```
const slop = 10 * time.Second
```

if your comment is correct.

Nonetheless, if it isn't, you should make
```
slop = 10 * time.Nanosecond
```
just to make it explicit what the units are. But I don't think you want a timeout of 10ns across the network :)